services/static.js

```diff
@@ -182,8 +219,21 @@ Static.getStaticImage = function(config) {
   if (config.logo !== undefined) {
     query.logo = String(config.logo);
   }
-  if (config.insertOverlayBeforeLayer !== undefined) {
-    query.before_layer = config.insertOverlayBeforeLayer;
+  if (config.before_layer !== undefined) {
+    query.before_layer = config.before_layer;
+  }
+  if (config.addlayer !== undefined) {
+    query.addlayer = config.addlayer;
+  }
+  if (config.setfilter !== undefined) {
+    query.setfilter = config.setfilter;
+  }
+  if (config.layer_id !== undefined) {
+    query.layer_id = config.layer_id;
+  }
+
+  if (config.setfilter !== undefined && config.layer_id === undefined) {
+    throw new Error('setfilter requires layer_id');
```

Comment:
This section is 💯, I would also add a check to make sure that `setfilter` and `addlayer` aren't being used simultaneously (as this is not currently supported by the API)

Reply:
Thank you! I'll also comb through the error messages to make sure everything is covered! https://docs.mapbox.com/api/maps/#static-images-api-errors